src/overlay005/event_check.c

```diff
@@ -0,0 +1,1095 @@
+#include "overlay005/ov5_021EA714.h"
+#include "overlay005/ov5_021EF4BC.h"
+#include "overlay005/ov5_021EFB0C.h"
+#include "overlay005/ov5_021F8370.h"
+#include "overlay006/ov6_02240C9C.h"
+#include "overlay006/ov6_02246BF4.h"
+#include "overlay008/ov8_02249960.h"
+#include "overlay009/ov9_02249960.h"
+#include "overlay023/ov23_02241F74.h"
+
+static BOOL FieldEventCheck_MapTransition(FieldSystem *fieldSystem, const FieldEventCheck *eventCheck);
+static u16 FieldEventCheck_TilePermissionsToEvent(FieldSystem *fieldSystem, u8 permissions);
+static BOOL FieldEventCheck_WildEncounter(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Step(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_PositionalEvent(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Transition(FieldSystem *fieldSystem, const int playerX, const int playerZ, const u8 curPermissions);
+static BOOL FieldEventCheck_Daycare(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Poison(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Safari(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_VsSeeker(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_PokeRadar(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Sign(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Repel(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_Friendship(FieldSystem *fieldSystem);
+static void FieldEventCheck_CalculateFriendship(FieldSystem *fieldSystem);
+static void FieldEventCheck_PlayerPos(const FieldSystem *fieldSystem, int *playerX, int *playerZ);
+static void FieldEventCheck_NextPos(const FieldSystem *fieldSystem, int *playerX, int *playerZ);
+static void FieldEventCheck_NextFacingPos(const FieldSystem *fieldSystem, int playerDir, int *playerX, int *playerZ);
+static u8 FieldEventCheck_CurrentTilePermissions(const FieldSystem *fieldSystem);
+static u8 FieldEventCheck_NextTilePermissions(const FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_MapConnection(const FieldSystem *fieldSystem, int playerX, int playerZ, Location *nextMap);
+static void FieldEventCheck_TrySetMapConnection(FieldSystem *fieldSystem);
+static BOOL FieldEventCheck_DistortionInteract(FieldSystem *fieldSystem, MapObject **object);
+static int FieldEventCheck_TrainerInfo(void);
+
+static void FieldEventCheck_Clear (FieldEventCheck *eventCheck)
+{
+    eventCheck->interact = FALSE;
+    eventCheck->endMovement = FALSE;
+    eventCheck->menu = FALSE;
+    eventCheck->registeredItem = FALSE;
+    eventCheck->sign = FALSE;
+    eventCheck->mapTransition = FALSE;
+    eventCheck->movement = FALSE;
+    eventCheck->unused1 = FALSE;
+    eventCheck->unused2 = FALSE;
+    eventCheck->unused3 = FALSE;
+    eventCheck->unused4 = FALSE;
+    eventCheck->unused5 = FALSE;
+    eventCheck->playerDir = DIR_NONE;
+    eventCheck->transitionDir = DIR_NONE;
+}
+
+void FieldEventCheck_Set (FieldEventCheck *eventCheck, FieldSystem *fieldSystem, u16 pressedKeys, u16 heldKeys)
```

Comment:
This name feels weird to me. It seems like this function is responsible for updating the check sets with player state. In that sense, it's more of an update than a set.